src/writer.js

```diff
@@ -0,0 +1,377 @@
+  return 1;
+}
+
+export function changeLeadResult (result) {
+  localStorage.removeItem(result.size);
+  document.getElementsByClassName(`size${result.size}`)[0].innerHTML = `<b>Time:</b> ${result.time}, <b>Turns:</b> ${result.turns} `;
+  localStorage.setItem(result.size, JSON.stringify(result));
+}
+
+export function showLeaderboard () {
+  if (localStorage.getItem(44)) {
+    let res1 = JSON.parse(localStorage.getItem(44));
+    document.getElementsByClassName('size44')[0].innerHTML = `<b>Time:</b> ${res1.time}, <b>Turns:</b> ${res1.turns} `;
+  }
+  if (localStorage.getItem(54)) {
+    let res2 = JSON.parse(localStorage.getItem(54));
+    document.getElementsByClassName('size54')[0].innerHTML = `<b>Time:</b> ${res2.time}, <b>Turns:</b> ${res2.turns} `;
+  }
+  if (localStorage.getItem(66)) {
+    let res3 = JSON.parse(localStorage.getItem(66));
+    document.getElementsByClassName('size66')[0].innerHTML = `<b>Time:</b> ${res3.time}, <b>Turns:</b> ${res3.turns} `;
+  }
+}
+
+export function saveGame (time, turns) {
+  let saveTime = document.getElementsByClassName('stopwatch')[0].value;
+  let saveTurns = document.getElementsByClassName('turnsCounter')[0].value;
+  let saveFieldSize = document.getElementsByClassName('fieldSelector')[0].value;
+  let currentField = document.getElementsByClassName('gameField')[0].innerHTML;
+
+  showMsg('<b>Saving...</b>');
+  setTimeout(function () {
+    document.getElementsByClassName('msgDiv')[0].remove();
+  }, 1000);
+
+  clearStopwatch();
+  clearTimeout(clocktimer);
+  init = 0;
+  stopTurns();
+  isFirstClick = true;
+  emptyCardNumber = 0;
+  Array.from(document.getElementsByClassName('card')).forEach((item) => {
+    if (item.classList.contains('invisible')) {
+      emptyCardNumber += 1;
+    }
+  });
+
+  let savedGame = {
+    'size': saveFieldSize,
+    'field': currentField,
+  }
+
+  localStorage.setItem('gameSave', JSON.stringify(savedGame));
+}
+
+export function loadGame (save) {
+  if (!save) {
+    showMsg('<b>No saved games found :(</b>');
+    setTimeout(function () {
+      document.getElementsByClassName('msgDiv')[0].remove();
+    }, 1000);
+    return;
+  }
+
+  let gameSave = JSON.parse(save);
+
+  document.getElementsByClassName('fieldSelector')[0].value = gameSave.size;
+  changeFieldSize();
+  document.getElementsByClassName('msgDiv')[0].innerHTML = '<p><b>Loading...</b></p>';
+
+  //loading saved field
+  setTimeout(function () {
+    document.getElementsByClassName('gameField')[0].innerHTML = gameSave.field;
+    emptyCardNumber = 0;
+    isFirstClick = false;
+    Array.from(document.getElementsByClassName('card')).forEach((item) => {
+      if (item.classList.contains('invisible')) {
+        emptyCardNumber += 1;
+      }
+    });
+  }, 1000);
+  isLoadedGame = true;
+}
+
+//Stopwatch initiaization and functions
+var base = 60;
+var clocktimer, dateObj, dh, dm, ds, ms;
+var readout = '';
+var h = 1,
+  m = 1,
+  tm = 1,
+  s = 0,
+  ts = 0,
```

Comment:
using global vars is not a good practice. it is always better to surroud it with closure/class.

Reply:
let me clarify: the best practice - split every single structure into modules and wrap them into classes (arrow functions). also it is better to implement a single pattern (e.g MVC/MVVM) to split rendering, logic and calls to the outer world. it makes code prettier and understandable for every team member. 

about var and const, let. since let and const were introduced it is preferable way to define variables. nevertheless you must also understand how var works, and why it acts sometimes in a weird way.

Reply:
e.g. what will be following code output and why?


const nums = [10, 12, 15, 21];
for (var i = 0; i < nums.length; i++) {
  setTimeout(function() {
    console.log('Index: ' + i + ', element: ' + arr[i]);
  }, 1000);
}

Reply:
Oh, I met questions of this kind, timeout opens a closure and var has one and only value for it, while let reopens it with new value every time (hope I said something reasonable). So the 'i' here will be '4' and there is no value for arr[4]

Reply:
@FinnYguY how can the code above print 10,12,15,21 numbers without using `let`?

Reply:
@avkulahina, that'll do the job: 
```
const nums = [10, 12, 15, 21];
for (var i = 0; i < nums.length; i++) {
  setTimeout(function(closurei) {
    return function() {
      console.log('Index: ' + closurei + ', element: ' + nums[closurei]);
    }
  }(i), 1000);
}
```

Reply:
Well done